### PR TITLE
Initial Setup and Initial Source Code

### PR DESCRIPTION
**Pull request by natec8**
+ Adds config files .editorconfig, .gitignore, .travis.yml
+ Adds phpunit.xml.dist
+ Adds composer files
+ Adds BaseSDK which takes care of requests and authentication with the
Cxense API
+ Adds unit tests
+ Adds Classes for cxense /traffic, /traffic/event, /traffic/custom
endpoints
+ Updates README.md

### CODE REVIEW

**Comment by natec8** [comment]
@janiv Could you take a look at what I've done so far and let me know if I'm on the right track?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+language: php
+php:
+- 5.5
```
Comment (janiv) [inline .travis.yml]: Let's add 5.6

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
...TkVuUVRjUzNyaDdxVGU0czFZeTFudXBuWEJXWld5bgpKZzEwbkd6S2pCVy9JSlB0aTl6MVpLTS9WMzQrazZxcGJacit4YU5zamYwUHZwenhsWU9odlFIeURJN1FoVFArCmQ3UUNkNXkvai9kZzFYaWRHdE9LRVhSN2lHUEpQNkFyM3pFRUxmM0U1U1RvSEZmQXVnMWo4Q3YxQW9HQkFPUDkKQXpLZnJ6ZFQxOS85K0FlelcyVUI2Zy9TZHBtT0pVNWI3bld2dFpieCtTVEJkZkw4eDYvdFU3L1BHMXZWdkRvVgp3ZkVpZ0xUVGJGcEI5dUwvUUQ5VXBqcFRCemVaNmREVmEvNWxxRzhKSFgwT3FFNVEvUVUwVzdLb3VNT1BEZU9aCkxneXVxUi9TTVplK0hqYTRuMHR6QWRyRDR3RmJrcm5zRUs0M3dxQVhBb0dBVCtZMDYrNTNhZWtBNlRmV2k0NGgKUk5kaDdYSHJCOEVFa1RRNm55VndjdC8vMDJUaTBjUzFVOHFPUC9hVXNFVHM1QThQZjZnalVEVGpmUDRDTEpZWQoxMHJPVkpaTzZDVGsyandSWDR4L1MyVzhPTVhBSW8vNExOUTAyU29aa0FXdFkrdldLd0dFL1VxTWpFQXdQcmFECmQ2VThBakdpY1J4a0lHajdmMWNQMHhrQ2dZRUFoQmVzeWFEeGNObjRJb1JRTjQzNGJ3NXI0WnpzTUUzR1daODAKQ3lXeHpYSUhDdiszNmdBQlpuSFg2ek1Bdy9XVEtBaEJKUWY3a0VsSzV3Ui9HY2dRSG1GYUtrSysxTDB0Vmx6QgpSRlQ4QU5GRUw1ZlZkZ0xBaGtwdXNwWEU3ekVzTitraFlDRzJSWVNRVFQzTzNaK2NVNmVmdEhONFMrb1ZtZmtuCk5QbURueU1DZ1lBZUFRb3N1dk1CZHpzOWRuT2lZbHAxT0RLZ3phUDJPOVZCcXFUWkxxN2tmZDViS0hKOXZ4YlIKVHNxSmpvVEswR3FueXR6aC9UakFOeXlMMldoU1pwbEtqUncwQUJ2WFRZNG5FMFg2ZTdSRnhCNGFZUEhxYWYrZApQTG1VMGk0ZTJvRzcwcHBEVVJjSTlqaGFMQ2syNzhHdmUxT3ZIMGM2UE1QQ3pmQSs0c2ZPL2c9PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
```
Comment (janiv) [inline .travis.yml]: I think this source_key is for installing fenix. I don't believe we need that there.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+language: php
+php:
+- 5.5
+cache:
+  directories:
+  - $HOME/.composer
+install:
+- composer install
+script:
+- ./vendor/bin/phpunit --coverage
```
Comment (janiv) [inline .travis.yml]: The --coverage option is for the gulp task in utah-new. To do coverage you'll need to use the phpunit config options instead and specify where to write the coverage logs.

You probably want `-coverage-clover <file>  Generate code coverage report in Clover XML format.`

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            "name": "Nate Christensen",
+            "email": "natec8@gmail.com"
+        }
+    ],
+    "require": {
+        "guzzlehttp/guzzle": "~6.1"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "~5.0",
+        "mockery/mockery": "~0.9.*",
```
Comment (janiv) [inline composer.json]: This line was giving me an error. I had to change it to "~0.9" instead of "~0.9.*"

File: src/BaseSDK.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                    'body' => json_encode($options),
+                    'headers' => [
+                        self::AUTH_HEADER => $this->generateAuthHeaderValue()
+                    ]
+                ]
+            );
+        } catch (GuzzleException $e) {
+            // Forward the exception on
+            throw $e;
+        }
```
Comment (janiv) [inline src/BaseSDK.php]: This try catch block seem unneeded

Comment (natec8) [inline src/BaseSDK.php]: ha good point, didn't think about that.

**Comment by janiv** [comment]
I was able to get data back from CXense. Is there a way to pass in filters, etc?

**Comment by janiv** [comment]
Okay, that's it for my comments. For merge I need:

* [ ] Fixes in .travis-ci
* [ ] Fixes in composer.json and thus composer.lock
* [ ] Remove the useless try/catch
* [ ] Answer question about how to use the cxense filters with this sdk

**Comment by natec8** [comment]
@janiv To send filters you would just send them in the options for example you can do something like this:

```php
$options = [
    'start' => strtotime('-1 week'),
    'stop' => time(),
    'fields' => [
        'events',
        'uniqueUsers',
        'urls',
        'activeTime',
        'sessionStarts',
        'sessionStops',
        'sessionBounces'
    ],
    'filters' => [
        [
            'type' => 'custom',
            'group' => 'grid_image',
            'items'  => ['no-image']
        ]
    ]
];

$results = $traffic->getData($options);
```

**Comment by natec8** [comment]
@janiv this is ready to review

**Pull request merged**